### PR TITLE
[SymbolGraph][CursorInfo] Add option to SourceKit's CursorInfo request to include the SymbolGraph JSON

### PR DESCRIPTION
**Pull request by nathawes**
Adds a new `key.retrieve_symbol_graph` option to the request. When set to 1 the JSON for a SymbolGraph containing a single node corresponding to the symbol at the cursor position will be included in the response under `key.symbol_graph`.
    
This extends the SymbolGraph with a new entry point to get a graph for a single symbol, and to additionally support type substitution to match the existing CursorInfo behavior (e.g. so that when invoked on `first` in `Array<Int>().first`, the type is reported as `Int?` rather than `Element?`).
    
Resolves rdar://problem/70551509

### CODE REVIEW

**Comment by nathawes** [comment]
@swift-ci Please Build Toolchain macOS Platform

**Comment by nathawes** [comment]
@swift-ci Please Build Toolchain macOS Platform

**Comment by nathawes** [comment]
@swift-ci please test

**Comment by nathawes** [comment]
@swift-ci Please Build Toolchain macOS Platform

File: lib/SymbolGraphGen/SymbolGraph.cpp [hunk 1]
```diff
@@ -499,7 +502,23 @@ void SymbolGraph::serialize(llvm::json::OStream &OS) {
...
+        M.getRequiredBystandersIfCrossImportOverlay(Declaring, Bystanders);
+      }
+
+      OS.attribute("name", ModuleName);
+      if (!Bystanders.empty()) {
+        OS.attributeArray("bystanders", [&](){
+          for (auto &Id: Bystanders)
+            OS.value(Id.str());
+        });
+      }
```
Comment (nathawes) [inline lib/SymbolGraphGen/SymbolGraph.cpp:521]: This part will go away once https://github.com/apple/swift/pull/34922 lands.

**Comment by nathawes** [comment]
@swift-ci please test

**Comment by nathawes** [comment]
@swift-ci Please Build Toolchain macOS Platform

**Comment by nathawes** [comment]
@swift-ci please test

**Comment by nathawes** [comment]
@swift-ci please test OS X Platform

**Comment by nathawes** [comment]
@swift-ci please test Linux Platform

**Review by QuietMisdreavus — APPROVED** [review]
Symbol graph stuff looks good! One little nit in the new entry point, otherwise i think this is nice.

**Comment by nathawes** [comment]
@swift-ci please test

**Comment by nathawes** [comment]
@akyrtzi I'm going to go ahead and merge this. If you have time to take a look I'll address any comments you have in a follow-up PR.

**Comment by nathawes** [comment]
@swift-ci Please Build Toolchain macOS Platform

**Pull request merged**